Criação das páginas CadastrarEditarCargo e AlocarCargo

NewtonJr5 opened the pull request.

NewtonJr5 committed:
Criação das páginas CadastrarEditarCargo e AlocarCargo

Itor-Carlos commented on src/pages/cargo/AlocarCargo.js:
- Trocar onde ta Grão por Cargo
- Os campos de cargo e funcionário na interface de AlocarCargo devem ser selects. Um deles vai mostrar todos os cargos no banco de dados e o outro todos os funcionários. Se quiser ajuda pra implementar, só chamar

NewtonJr5 committed:
Correções.

NewtonJr5 commented:
Corrigido.

Itor-Carlos commented on src/pages/cargo/AlocarCargo.js:
* Os dados enviados para o backend devem ser o id do Cargo e o id do Funcionário.
* Ao acessar a página dá um erro de imediato devido a falta do options nos campos que são select. Cargo ainda não da pra mostrar todos devido a rota não ter sido implementada. Entretanto tu pode fazer isso com o campo de funcionário já. Se tiver dúvida é só olhar a página de AlocarTarefa.

Print abaixo do erro mencionado anteriormente:

![image](https://github.com/user-attachments/assets/7fccfeab-eaf4-48c5-b69a-4228c7d6510c)

NewtonJr5 committed:
Correção de erros.

Estou com duvida se as rotas estão corretas.
Também encontrei problemas com a função map, mas consegui resolver, só não sei se foi da melhor forma.

Itor-Carlos commented:
Qual problema com a função map?

Itor-Carlos commented on src/pages/cargo/AlocarCargo.js:
No handleSubmit de Alocar Cargo não é necessário passar as informações com o values.nome e values.funcionario. Basta passar a rota e os valores no value.

NewtonJr5 commented:
O map estava sendo aplicado em um elemento undefined. Ai dava erro.

NewtonJr5 commented:
Onde que precisa enviar o idfuncionário e o idcargo que voçê falou la em cima?

Itor-Carlos commented:
> Onde que precisa enviar o idfuncionário e o idcargo que voçê falou la em cima?

- Dificilmente a gente vai ter uma interface pra editar essa parte de AlocarCargo. Então essa parte do isEditing pode ser removido
- Os valores de idfuncionario e idcargo que eu falei na sua interface são respectivamente: funcionario e nome. Me expressei mal

- Tire essa parte da edição que eu aceito o PR

NewtonJr5 committed:
Remoção da variável isEditing

Itor-Carlos committed:
Merge branch 'main' into modulo_cargo
